queries/cpp/highlights.scm

```diff
@@ -4,7 +4,12 @@
  (#match? @field "(^_|^m_|_$)"))
 
 (parameter_declaration
-  declarator: (reference_declarator) @parameter)
+  declarator: (reference_declarator) @reference.declarator (#set! "priority" 101)) ; priority to override operator
```

Comment:
same as with c pointer to cpp reference.

Reply:
again it is not worth to create a new highlight category out of the nothing. The subscoping mechanism is thought to have as few groups as possible on the top-level and go more specific in the subscope. In each case, the reference declarator and pointer declarator should go into the same category. Such a change should probably be then applied to Rust and go also. So it might be worth to discuss this topic in a separate PR. In each case it should be in sync with #3874